src/constraint.rs

```diff
@@ -73,6 +78,79 @@ where
     Box::new(Impossible { constraint })
 }
 
+pub fn implies<Var, Value>(
```

Comment:
why are these `pub` and not `pub(crate)`?

Reply:
The implies constraint is useful not only for groundedness check but also type checking in general (e.g., type-level literals). We want users to use these constraints for their primitives.